executor/adapter.go

```diff
@@ -568,6 +569,32 @@ func (a *ExecStmt) handlePessimisticDML(ctx context.Context, e Executor) error {
 	}
 }
 
+// GetTimestampWithRetry tries to get timestamp using retry and backoff mechanism
+func (a *ExecStmt) GetTimestampWithRetry(ctx context.Context) (uint64, error) {
+	tsoMaxBackoff := 15000
+	bo := tikv.NewBackoffer(context.Background(), tsoMaxBackoff)
+	if span := opentracing.SpanFromContext(ctx); span != nil && span.Tracer() != nil {
+		span1 := span.Tracer().StartSpan("ExecStmt.GetTimestampWithRetry", opentracing.ChildOf(span.Context()))
+		defer span1.Finish()
+		ctx = opentracing.ContextWithSpan(ctx, span1)
+	}
+	for {
+		ts, err := a.Ctx.GetStore().GetOracle().GetTimestamp(ctx)
+		// mock get ts fail
+		failpoint.Inject("ExecStmt get ts error", func() (uint64, error) {
```

Comment:
where is the test case that used the failpoint?

Reply:
I'll add some case cover this branch.  Oops, CI seems to have some problem, do we release test using this branch? @jackysp

Reply:
"ExecStmt get ts error" is not a good name.
You can grep the code to see how to use `failpoint.Inject`

Reply:
Roger, refactored and case added